# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../musl/pkg/gdb/gdb-linux_nat.patch
# Copyright (C) 2014 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

This is a patch from sabotage-linux
https://raw.github.com/sabotage-linux/sabotage/master/KEEP/gdb-linux_nat.patch

the first chunk fixes build errors,
the non_stop part below fixes thread debugging being available
by default without setting any breakpoints or usage of libthread_db.

--- gdb-7.4.org/gdb/linux-nat.c	2013-08-10 05:24:24.651000003 +0000
+++ gdb-7.4/gdb/linux-nat.c	2013-08-10 05:25:50.966000003 +0000
@@ -71,6 +71,14 @@
 # endif
 #endif /* HAVE_PERSONALITY */
 
+#ifndef __SIGRTMIN
+#define __SIGRTMIN SIGRTMIN
+#endif
+
+#ifndef W_STOPCODE
+#define W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
+
 /* This comment documents high-level logic of this file.
 
 Waiting for events in sync mode
@@ -2265,7 +2273,7 @@
 		status = 0;
 	    }
 
-	  if (non_stop)
+	  if (1)
 	    {
 	      /* Add the new thread to GDB's lists as soon as possible
 		 so that:
